packages/sdk-core/src/model/http/SubmissionUrlInformation.ts

```diff
@@ -0,0 +1,71 @@
+export class SubmissionUrlInformation {
+    private static SUBMIT_PREFIX = 'submit.backtrace.io/';
+
+    /**
+     * Convert url/token from credentials to JSON submission URL
+     * @param url credentials URL
+     * @param token credentials token
+     * @returns JSON submissionURL
+     */
+    public static toJsonReportSubmissionUrl(url: string, token?: string): string {
+        // if the token doesn't exist - use URL
+        if (!token) {
+            return url;
+        }
+
+        // if the url points to submit, we should always use it without any modifications
+        if (url.includes(this.SUBMIT_PREFIX)) {
+            return url;
+        }
+
+        // if the URL has token in the URL, the user probably added a token once again
+        // in this case, don't do anything
+        if (url.indexOf(token) !== -1) {
+            return url;
+        }
+
+        return new URL(`/post?format=json&token=${token}`, url).href;
```

Comment:
In that case, I would create the URL, and in `searchParams` set the `"token"` query param to `token`.